### ISSUE TITLE
[Feature Request]: The length of line after spaces inserted can be aligned to [Space count]

### ISSUE THREAD

**Issue by ETOgaosion** [jrymk/restore-tab-key#8]
In some cases, users who want to use spaces to replace tabs, meanwhile can have same align behavior like tab, like vscode does.

take `Space count=4` for example:

```py
# before
abc[tab]
# after
abc
    ^ cursor
```

rather than:

```py
abc
       ^ cursor
```

So the `Space count` option in plugin setting will be defined as `Max Space Count`.

This issue may improve user experience.

**Comment by ETOgaosion** [comment]
Yep! Only code blocks may need this `tab` check to get perfect alignment, other type of input blocks may not need in common users' workflow.

**Comment by jrymk** [comment]
I will add code block check when I have free time/when I feel like so.
My point is that people that use tabs in Obsidian most likely only press it in code blocks, as tabbing in other places doesn't really do anything due to the nature of Markdown.
So I think it is fine to leave the align feature on regardless of code block, for now at least. 
I'll see if there's an efficient way of detecting code blocks without checking every line from the beginning of the document. I'll close this issue for now.
